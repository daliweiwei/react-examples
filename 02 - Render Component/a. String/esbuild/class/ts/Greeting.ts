/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

import { Component } from 'react';

export default class Greeting extends Component {
  render(): string {
    return 'Hello World!';
  }
}
