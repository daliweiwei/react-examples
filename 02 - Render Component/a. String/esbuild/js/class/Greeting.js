/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

import { Component } from 'react';

class Greeting extends Component {
  render() {
    return 'Hello World!';
  }
}

export default Greeting;
