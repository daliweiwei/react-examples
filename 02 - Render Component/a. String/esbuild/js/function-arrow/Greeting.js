/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

const Greeting = () => 'Hello World!';

export default Greeting;
