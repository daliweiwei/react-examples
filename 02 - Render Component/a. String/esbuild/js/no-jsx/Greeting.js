/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

function Greeting() {
  return 'Hello World!';
}

export default Greeting;
