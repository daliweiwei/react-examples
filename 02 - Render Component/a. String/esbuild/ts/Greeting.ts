/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

export default function Greeting() {
  return 'Hello World!';
}
