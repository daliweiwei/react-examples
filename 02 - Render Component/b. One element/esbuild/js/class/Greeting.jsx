/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

import { Component } from 'react';

class Greeting extends Component {
  render() {
    return <h1>Hello World!</h1>;
  }
}

export default Greeting;
