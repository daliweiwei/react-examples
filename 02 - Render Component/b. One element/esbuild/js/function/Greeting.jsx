/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

function Greeting() {
  return <h1>Hello World!</h1>;
}

export default Greeting;
