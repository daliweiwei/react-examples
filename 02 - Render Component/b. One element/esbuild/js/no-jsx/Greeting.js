/*! European Union Public License version 1.2 !*/
/*! Copyright Â© 2016 Rick Beerendonk          !*/

import React from 'react';

function Greeting() {
  return React.createElement('h1', null, 'Hello World!');
}

export default Greeting;
