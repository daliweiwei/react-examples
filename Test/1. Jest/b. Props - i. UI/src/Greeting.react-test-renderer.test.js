import React from 'react';
import renderer from 'react-test-renderer';
import Greeting from './Greeting';

/// React Test Renderer ///

describe('Greeting', () => {
  it('throws when no name is provided (react-test-renderer)', () => {
    expect(() => {
      renderer.create(<Greeting />);
    }).toThrow();
  });

  it('renders hello world inline snapshot when name is world (react-test-renderer)', () => {
    const component = renderer.create(<Greeting name="World" />);
    let tree = component.toJSON();
    expect(tree).toMatchInlineSnapshot(`
  <h1
    className="greeting"
  >
    Hello 
    World
    !
  </h1>
  `);
  });

  it('renders hello world snapshot when name is world (react-test-renderer)', () => {
    const component = renderer.create(<Greeting name="World" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
