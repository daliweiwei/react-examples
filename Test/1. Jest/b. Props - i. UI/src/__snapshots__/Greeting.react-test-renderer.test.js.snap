// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting renders hello world snapshot when name is world (react-test-renderer) 1`] = `
<h1
  className="greeting"
>
  Hello 
  World
  !
</h1>
`;
